lua/apisix/core.lua

```diff
@@ -39,3 +43,15 @@ return {
     http     = require("apisix.core.http"),
     tablepool= require("tablepool"),
 }
+
+
+function _M.discovery ()
```

Comment:
we'd better keep old-style `local discovery  = require("apisix.core").discovery`

```lua
_M.discovery = xxx
```

Reply:
@qiujiayu please take a look

Reply:
had move to `apisix. discovery .init`.